plugins/opener/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+
+mod commands;
+mod error;
+mod open;
+mod reveal_item_in_dir;
+mod scope;
+mod scope_entry;
+
+pub use error::Error;
+type Result<T> = std::result::Result<T, Error>;
+
+pub use open::{open_path, open_url};
+pub use reveal_item_in_dir::reveal_item_in_dir;
+
+pub struct Opener<R: Runtime> {
+    // we use `fn() -> R` to slicence the unused generic error
+    // while keeping this struct `Send + Sync` without requiring `R` to be
+    #[cfg(not(mobile))]
+    _marker: std::marker::PhantomData<fn() -> R>,
+    #[cfg(mobile)]
+    mobile_plugin_handle: PluginHandle<R>,
+}
+
+impl<R: Runtime> Opener<R> {
+    /// Open a url with a default or specific program.
+    ///
+    /// ## Platform-specific:
+    ///
+    /// - **Android / iOS**: Always opens using default program.
+    #[cfg(desktop)]
+    pub fn open_url(&self, url: impl Into<String>, with: Option<impl Into<String>>) -> Result<()> {
+        crate::open::open(url.into(), with.map(Into::into)).map_err(Into::into)
+    }
+
+    /// Open a url with a default or specific program.
+    ///
+    /// ## Platform-specific:
+    ///
+    /// - **Android / iOS**: Always opens using default program.
+    #[cfg(mobile)]
+    pub fn open_url(&self, url: impl Into<String>, _with: Option<impl Into<String>>) -> Result<()> {
+        self.mobile_plugin_handle
+            .run_mobile_plugin("open", url.into())
+            .map_err(Into::into)
+    }
+
+    /// Open a path with a default or specific program.
+    ///
+    /// ## Platform-specific:
+    ///
+    /// - **Android / iOS**: Always opens using default program.
+    #[cfg(desktop)]
+    pub fn open_path(
+        &self,
+        path: impl Into<String>,
+        with: Option<impl Into<String>>,
+    ) -> Result<()> {
+        crate::open::open(path.into(), with.map(Into::into)).map_err(Into::into)
+    }
+
+    /// Open a path with a default or specific program.
+    ///
+    /// ## Platform-specific:
+    ///
+    /// - **Android / iOS**: Always opens using default program.
+    #[cfg(mobile)]
+    pub fn open_path(
+        &self,
+        path: impl Into<String>,
+        _with: Option<impl Into<String>>,
+    ) -> Result<()> {
+        self.mobile_plugin_handle
+            .run_mobile_plugin("open", path.into())
+            .map_err(Into::into)
+    }
+
+    pub fn reveal_item_in_dir<P: AsRef<Path>>(&self, p: P) -> Result<()> {
+        crate::reveal_item_in_dir::reveal_item_in_dir(p)
+    }
+}
+
+/// Extensions to [`tauri::App`], [`tauri::AppHandle`], [`tauri::WebviewWindow`], [`tauri::Webview`] and [`tauri::Window`] to access the opener APIs.
+pub trait OpenerExt<R: Runtime> {
+    fn opener(&self) -> &Opener<R>;
+}
+
+impl<R: Runtime, T: Manager<R>> crate::OpenerExt<R> for T {
+    fn opener(&self) -> &Opener<R> {
+        self.state::<Opener<R>>().inner()
+    }
+}
+
+/// Initializes the plugin.
+pub fn init<R: Runtime>() -> TauriPlugin<R> {
+    Builder::new("opener")
+        .js_init_script(include_str!("init-iife.js").to_string())
```

Comment:
I still feel like this should be opt-in or at least be optional, and in that case, we'll need this to be a builder or it will be a breaking change when we add it

Reply:
added